packages/service-library/src/servicelib/fastapi/long_running_tasks/_routes.py

```diff
@@ -42,22 +41,17 @@ async def get_task_result(
 ) -> TaskResult:
     assert request  # nosec
 
-    remove_task = True
-
-    try:
-        task_result = task_manager.get_result(task_id=task_id)
-    except TaskNotCompletedError:
-        remove_task = False
-        raise
-    finally:
-        if remove_task:
-            await task_manager.remove(task_id, reraise_errors=False)
+    task_result = task_manager.get_result(task_id=task_id)
+    await task_manager.remove(task_id, reraise_errors=False)
```

Comment:
So from what I understand this is equivalent. Am I getting it right? After reading the code again I think the task always gets removed after the results is recovered and if the task is not completed it would still stay there.

Reply:
right. I think this is exactly the same. and from pytests I could see that the last part was not properly covered also.
originaly I thought the remove was a query parameter but it is not. so that way it is the same.

Reply:
I'm fine with this. Think I managed to confuse myself somehow here. Glad it is simpler.